src/main/java/uk/gov/ch/developer/docs/controller/developer/UserCallbackController.java

```diff
@@ -35,17 +35,16 @@ public String getCallback(@RequestParam("state") String state,
             @RequestParam("code") String code, final HttpServletRequest httpServletRequest) {
         LOGGER.trace("Code:" + code);
         LOGGER.trace("State:" + state);
+        final Session chSession = (Session) httpServletRequest
+                .getAttribute(SessionHandler.CHS_SESSION_REQUEST_ATT_KEY);
 
         final String returnedNonce = getNonceFromState(state);
-        if (!oauth.oauth2VerifyNonce(returnedNonce)) {
+        if (!oauth.oauth2VerifyNonce(returnedNonce, chSession)) {
```

Comment:
Rather than passing the session through, it would be better to call `getSessionNonce(chSession)` outside then pass the result into `oauth2VerifyNonce(...)`

On the lines of:

```
final String returnedNonce = getNonceFromState(state);
final String sessionNonce = oauth2.getAndDestroySessionNonce(chSession);
if (!oauth.oauth2VerifyNonce(returnedNonce, sessionNonce)) {

```

Reply:
I disagree. I'm working on a ticket to remove generating nonces from the signin controller after all.